driver/csiplugin/controllerserver.go

```diff
@@ -1974,16 +1975,20 @@ func (cs *ScaleControllerServer) DeleteVolume(ctx context.Context, req *csi.Dele
 	relPath = strings.Trim(relPath, "!/")
 	isPvcFromSnapshot := false
 	var shallowCopyRefPath string
-	if volumeIdMembers.VolType == FILE_SHALLOWCOPY_VOLUME {
-		if relPath != "" && strings.Contains(relPath, ".snapshots") {
-			volPath := strings.Split(relPath, "/")
-			if len(volPath) > 2 {
-				if volPath[1] == ".snapshots" {
-					isPvcFromSnapshot = true
-					shallowCopyRefPath = fmt.Sprintf("%s/%s", volPath[0], volPath[2])
-				}
-			}
-		}
+	var snapshotName string
+  	var independentFileset string
+	if volumeIdMembers.VolType == FILE_SHALLOWCOPY_VOLUME{
+    		if relPath != "" && strings.Contains(relPath, ".snapshots"){
+      			volPath := strings.Split(relPath, "/")
+        		if len(volPath) > 2{
```

Comment:
can we add else when length is less than or equal to 2 with error or just log, or what will happen when if doesn't meet

Reply:
Done